### ISSUE TITLE
JdbcOAuth2AuthorizationService does not work with postgresql

### ISSUE THREAD

**Issue by gabydinu** [spring-projects/spring-authorization-server#478]
**Describe the bug**
When using JdbcOAuth2AuthorizationService with PostgreSQL database, blob columns in table should be defined as bytea.
When a request is provided to exchange authorization_code to authorization_token, method findByToken is called.

```
 else if (OAuth2ParameterNames.CODE.equals(tokenType.getValue())) {
			parameters.add(new SqlParameterValue(Types.BLOB, token.getBytes(StandardCharsets.UTF_8)));
			return findBy(AUTHORIZATION_CODE_FILTER, parameters);


```
Which is calling _findBy_:
```
PreparedStatementSetter pss = new ArgumentPreparedStatementSetter(parameters.toArray());
		List<OAuth2Authorization> result = this.jdbcOperations.query(LOAD_AUTHORIZATION_SQL + filter, pss, this.authorizationRowMapper);
		return !result.isEmpty() ? result.get(0) : null;
```

This way in PGPrepareStatement will break as it receive byte[] :
```
case Types.BLOB:
        if (in instanceof Blob) {
          setBlob(parameterIndex, (Blob) in);
        } else if (in instanceof InputStream) {
          long oid = createBlob(parameterIndex, (InputStream) in, -1);
          setLong(parameterIndex, oid);
        } else {
          throw new PSQLException(
              GT.tr("Cannot cast an instance of {0} to type {1}",
                  in.getClass().getName(), "Types.BLOB"),
              PSQLState.INVALID_PARAMETER_TYPE);
        }
```
** Posible fix **
defining _findBy_ as:
```
try (LobCreator lobCreator = this.lobHandler.getLobCreator()) {
			PreparedStatementSetter pss = new JdbcOAuth2AuthorizationService.LobCreatorArgumentPreparedStatementSetter(lobCreator,
					parameters.toArray());
			List<OAuth2Authorization> result = this.jdbcOperations.query(LOAD_AUTHORIZATION_SQL + filter, pss, this.authorizationRowMapper);
			return !result.isEmpty() ? result.get(0) : null;
		}
```

 
**To Reproduce**
Use JdbcOAuth2AuthorizationService with PostgreSQL and try to do an authorization_code flow.

**Comment by ovidiupopa91** [comment]
Hi. This was fixed in [455](https://github.com/spring-projects/spring-authorization-server/issues/455). commit [](https://github.com/spring-projects/spring-authorization-server/commit/33bac0f7c214bf63a36639084551b00642d341f9).

It will be released in `0.2.1`

**Comment by gabydinu** [comment]
thank you
